reactor-core/src/test/java/reactor/core/publisher/FluxReplayTest.java

```diff
@@ -331,27 +311,36 @@ protected void hookOnNext(Integer value) {
 		assertThat(requests).isEmpty();
 
 		replay.subscribe(fiveThenEightSubscriber);
+		replay.subscribe(sevenThenEightSubscriber);
 		replay.connect();
 
-		assertThat(requests).containsExactly(5L);
+		assertThat(requests).startsWith(7L).doesNotContain(5L);
 	}
 
 	@Test
-	public void minimalInitialRequestWithUnboundedSubscriber() {
+	public void multipleEarlySubscribersPropagateTheirLateRequests() {
 		List<Long> requests = new ArrayList<>();
-		BaseSubscriber<Integer> fiveThenEightSubscriber = new BaseSubscriber<Integer>() {
-			@Override
-			protected void hookOnSubscribe(Subscription subscription) {
-				subscription.request(5);
-			}
+		TwoRequestsSubscriber fiveThenEightSubscriber = new TwoRequestsSubscriber(5, 8);
```

Comment:
idea: since the subscribers are not used by reference anywhere except `subscribe`, we can remove the local variables and inline them, so that we don't need to encode the numbers in the var's name:
```java
replay.take(13).subscribe(new TwoRequestsSubscriber(5, 8));
replay.subscribe(new TwoRequestsSubscriber(7, 8));
```